packages/desktop-client/src/components/transactions/MobileTransaction.js

```diff
@@ -295,15 +316,31 @@ class TransactionEditInner extends PureComponent {
   };
 
   onDelete = () => {
-    this.props.onDelete();
+    let onConfirmDelete = () => {
+      this.props.onDelete();
+
+      const { transactions } = this.state;
+      const [transaction, ..._childTransactions] = transactions;
+      const { account: accountId } = transaction;
+      if (accountId) {
+        this.props.navigate(`/accounts/${accountId}`, { replace: true });
+      } else {
+        this.props.navigate(-1);
+      }
+    };
 
     const { transactions } = this.state;
     const [transaction, ..._childTransactions] = transactions;
-    const { account: accountId } = transaction;
-    if (accountId) {
-      this.props.navigate(`/accounts/${accountId}`, { replace: true });
+
+    if (transaction.reconciled) {
+      this.props.pushModal('confirm-transaction-edit', {
+        onConfirm: () => {
+          onConfirmDelete();
+        },
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
        onConfirm: onConfirmDelete,
```